src/Raven.Server/Documents/Sharding/Handlers/Processors/Indexes/ShardedIndexHandlerProcessorForTerms.cs

```diff
@@ -0,0 +1,85 @@
+namespace Raven.Server.Documents.Sharding.Handlers.Processors.Indexes
+{
+    internal class ShardedIndexHandlerProcessorForTerms : AbstractIndexHandlerProcessorForTerms<ShardedDatabaseRequestHandler, TransactionOperationContext>
+    {
+        public ShardedIndexHandlerProcessorForTerms([NotNull] ShardedDatabaseRequestHandler requestHandler) : base(requestHandler, requestHandler.ContextPool)
+        {
+        }
+
+        protected override async ValueTask<TermsQueryResultServerSide> GetTerms(TransactionOperationContext context, string indexName, string field, string fromValue, int pageSize)
+        {
+            var op = new ShardedGetTermsOperation(RequestHandler, indexName, field, fromValue, pageSize);
+            var result = await RequestHandler.ShardExecutor.ExecuteParallelForAllAsync(op);
+
+            return result;
+        }
+
+
+        private readonly struct ShardedGetTermsOperation : IShardedOperation<TermsQueryResultServerSide>
+        {
+            private readonly ShardedDatabaseRequestHandler _handler;
+            private readonly string _indexName;
+            private readonly string _field;
+            private readonly string _fromValue;
+            private readonly int _pageSize;
+
+            public ShardedGetTermsOperation(ShardedDatabaseRequestHandler handler, string indexName, string field, string fromValue, int pageSize)
+            {
+                _handler = handler;
+                _indexName = indexName;
+                _field = field;
+                _fromValue = fromValue;
+                _pageSize = pageSize;
+            }
+
+            public HttpRequest HttpRequest => _handler.HttpContext.Request;
+
+            public TermsQueryResultServerSide Combine(Memory<TermsQueryResultServerSide> results)
+            {
+                var pageSize = _pageSize;
+                var terms = new HashSet<string>();
+                foreach (var res in _handler.DatabaseContext.Streaming.CombinedResults(results, r => r.Terms, TermsComparer.Instance))
+                {
+                    if (terms.Add(res.Item))
+                        pageSize--;
+
+                    if (pageSize <= 0)
+                        break;
+                }
+
+                return new TermsQueryResultServerSide { IndexName = results.Span[0].IndexName, ResultEtag = results.Span[0].ResultEtag, Terms = terms };
+            }
+
+            public RavenCommand<TermsQueryResultServerSide> CreateCommandForShard(int shard) => new GetIndexTermsCommand(indexName: _indexName, field: _field, _fromValue, _pageSize);
+        }
+
+        public class TermsComparer : Comparer<ShardStreamItem<string>>
+        {
+            public override int Compare(ShardStreamItem<string> x,
+                ShardStreamItem<string> y)
+            {
+                if (x == null)
+                    return -1;
+
+                if (y == null)
+                    return 1;
+
+                return string.CompareOrdinal(x.Item, y.Item);
```

Comment:
I think 
`return string.CompareOrdinal(x?.Item, y?.Item);` 
should work as well

Reply:
done